### ISSUE TITLE
Customer Search is Still Broken

### ISSUE THREAD

**Issue by CodeConnectPtyLtd** [woocommerce/woocommerce-admin#7588]
### Describe the bug
Customer report search functionality returns no data when there are clearly results that should be returned.

### To Reproduce

1. Go to WooCommerce → Customers
2. In the search bar, type part (or all) of the name of a customer
3. Click the resulting "All customers with names that include {whatever you typed}"
4. See a blank table with the message No data to display

### Actual behaviour:
https://streamable.com/f753dc
As documented in closed ticket below however the results returned appears to be intermittently retuning no data or the correct results.  I can reproduce this across several installations.  Previously closed issue that also relates.
https://github.com/woocommerce/woocommerce-admin/issues/7486

### Expected behaviour
Expected  behaviour would be consistency of the search results returned in a case insensitive manner.

### Desktop (please complete the following information):

* OS: Windwos 10
* Browser Brave
* Version 1.28.106

**Comment by ilyasfoo** [comment]
Thank you so much for taking the time to report this bug @CodeConnectPtyLtd! Your video attachment explained the bug very well.

I was able to reproduce the bug. It seems there could be some issues in rendering results after a search because I've also monitored the networks tab to see the results from ajax and indeed it was returning them (even when the table says "No data to display").

I do not think the case sensitivity is related to the bug, I was able to obtain the result after searching for the same text twice:

![image](https://user-images.githubusercontent.com/3747241/131615083-9d55d908-2fad-4d1f-a5af-616512bd2ffe.png)


Also, note that this issue is similar to another one caught in https://github.com/woocommerce/woocommerce-admin/issues/7258. I suspect they are caused by the same bug.
